nautobot/dcim/templates/dcim/device.html

```diff
@@ -222,6 +222,24 @@
                                             {{ object.software_version|hyperlinked_object }}
                                         </td>
                                     </tr>
+                                    <tr>
+                                        <td>Software Image files</td>
+                                        <td>
+                                            {% with software_image_file_count=object.software_image_files.count %}
+                                                {% if software_image_file_count %}
+                                                <a href="{% url 'dcim:softwareimagefile_list' %}?devices={{ object.pk }}">{{ software_image_file_count }}</a>*
+                                                {% else %}
+                                                    {% with software_version_image_file_count=object.software_version.software_image_files.count %}
```

Comment:
Is this correct, or should we inherit from `object.device_type.software_image_files` instead?

Reply:
That gets into the inheritance conversation. I think right now device type <-> software image just determines what images are compatible with a device type. The device -> software version <-> software image is what determines what is (or should be) running on that device, unless you override it with device <-> software image. 😵